src/main/java/org/spongepowered/common/mixin/core/inventory/ContainerMixin.java

```diff
@@ -119,7 +120,8 @@
     //private boolean postPreCraftEvent = true; // used to prevent multiple craft events to fire when setting multiple slots simultaneously
     private List<SlotTransaction> impl$capturedSlotTransactions = new ArrayList<>();
     private List<SlotTransaction> impl$capturedCraftShiftTransactions = new ArrayList<>();
-    private List<SlotTransaction> impl$capturedCraftPreviewTransactions = new ArrayList<>();
+    @Nullable private SlotTransaction impl$capturedCraftPreviewTransaction;
+    private static int impl$numTransactionErrorsLogged = 0;
```

Comment:
Static fields should be in their own body imo. Make it well aware they are static.